Realm/Realm/Realm.cs

```diff
@@ -247,22 +247,9 @@ public Session SyncSession
                     throw new NotSupportedException("Realm.SyncSession is only valid for synchronized Realms (i.e. ones that are opened with FlexibleSyncConfiguration or PartitionSyncConfiguration).");
                 }
 
-                if (_sessionRef is null || !_sessionRef.TryGetTarget(out var session) || session.IsClosed)
-                {
-                    var sessionHandle = SharedRealmHandle.GetSession();
-                    session = new Session(sessionHandle);
-
-                    if (_sessionRef is null)
-                    {
-                        _sessionRef = new WeakReference<Session>(session);
-                    }
-                    else
-                    {
-                        _sessionRef.SetTarget(session);
-                    }
-                }
+                _sessionProvider ??= new SessionProvider();
```

Comment:
You can pass the handle in the provider ctor as it doesn't appear like the provider is shared between different realms.

Reply:
Done